integration-tests-support/aws2/src/main/java/org/apache/camel/quarkus/test/support/aws2/BaseAws2Resource.java

```diff
@@ -0,0 +1,96 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.test.support.aws2;
+
+import javax.enterprise.event.Observes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.quarkus.runtime.ShutdownEvent;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+
+/**
+ * Parent for testing resources for aws2 extensions.
+ *
+ * This class adds endpoints to force default credentials to be used + provides protected varialbe useDefaultCredentials
+ * to be used in inherited classes.
+ *
+ * Rest endpoints are expected by test classes.
+ */
+public class BaseAws2Resource {
+
+    private static final Logger LOG = Logger.getLogger(BaseAws2Resource.class);
+
+    protected boolean useDefaultCredentials;
+
+    private final String serviceName;
+
+    private boolean clearAwsredentials;
+
+    public BaseAws2Resource(String serviceName) {
+        this.serviceName = serviceName;
+    }
+
+    @Path("/setUseDefaultCredentialsProvider")
+    @POST
+    public Response setUseDefaultCredentials(boolean useDefaultCredentialsProvider) throws Exception {
+        this.useDefaultCredentials = useDefaultCredentialsProvider;
+        return Response.ok().build();
+    }
+
+    @Path("/initializeDefaultCredentials")
+    @POST
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response initializeDefaultCredentials(boolean initialize) throws Exception {
+        if (initialize) {
+            //set flag to clear at the end
+            clearAwsredentials = true;
+            LOG.debug(
+                    "Setting both System.properties `aws.secretAccessKey` and `aws.accessKeyId` to cover defaultCredentialsProviderTest.");
+            //defaultCredentials provider gets the credentials from fixed location. One of them is system.properties,
+            //therefore to succeed the test, system.properties has to be initialized with the values from the configuration
+            Aws2Helper.setAwsSystemCredentials(
+                    ConfigProvider.getConfig().getValue("camel.component.aws2-" + serviceName + ".access-key", String.class),
+                    ConfigProvider.getConfig().getValue("camel.component.aws2-" + serviceName + ".secret-key", String.class));
+
+        } else {
+            LOG.debug("Clearing both System.properties `aws.secretAccessKey` and `aws.accessKeyId`.");
+            Aws2Helper.clearAwsSystemCredentials();
+            clearAwsredentials = false;
+        }
+
+        return Response.ok().build();
+    }
+
+    /**
+     * Listeners ensures, that system credentials are cleared at the end of the lifecycle.
+     * Tests are clearing them by itself, this is just a precaution.
+     */
```

Comment:
Are we also clearing the properties when an unexpected exception is thrown during a test ?

Reply:
I thought that this listener should be called also with unexpected shutdown. Therefore it should clear properties for all cases. Am I missing here something? (https://quarkus.io/guides/lifecycle#listening-for-startup-and-shutdown-events)

Reply:
I can hook another safety clear into [Aws2TestResource.java](https://github.com/apache/camel-quarkus/blob/main/integration-tests-support/aws2/src/main/java/org/apache/camel/quarkus/test/support/aws2/Aws2TestResource.java#L109). That could help, what do you think, @aldettinger ?

Reply:
As far as I get, the `onStop` listener is called when quarkus shutdown.

The question would be more like, do we have situations where we set properties, then exception is thrown without cleanup. And then, a next test would fail because the properties was not cleaned up.

Reply:
Ok, it was solved in a comment below. There is a finally block in the test that would restore the properties.